compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1069,9 +1069,21 @@ object Parsers {
         SingletonTypeTree(literal(negOffset = start))
       }
       else if (in.token == USCORE) {
+        if (ctx.settings.strict.value) {
+          deprecationWarning(em"`_` is deprecated for wildcard arguments of types: use `?` instead")
+          patch(source, Span(in.offset, in.offset + 1), "?")
+        }
         val start = in.skipToken()
         typeBounds().withSpan(Span(start, in.lastOffset, start))
       }
+      else if (isIdent(nme.?)) {
```

Comment:
Should it be tpnme ?

Reply:
No, the idents coming form the token stream are always TermIdents.